File: crypto/src/main/java/io/oversec/one/crypto/gpg/GpgPreferences.java
/*
 * Copyright (C) 2012-2014 Dominik Schürmann <dominik@dominikschuermann.de>
 * Copyright (C) 2010-2014 Thialfihar <thi@thialfihar.org>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package io.oversec.one.crypto.gpg;


import android.annotation.SuppressLint;
import android.content.Context;
import android.content.SharedPreferences;


@SuppressLint("CommitPrefEdits")
public class GpgPreferences {

    private static final String PREF_PGP_OWN_PUBLIC_KEY_ID = "pgp_own_public_key_id";


    private static GpgPreferences mGpgPreferences;
    private SharedPreferences mSharedPreferences;

    private static String PREF_FILE_NAME = GpgPreferences.class.getSimpleName();


    public static synchronized GpgPreferences getPreferences(Context context) {
        if (mGpgPreferences == null) {
            mGpgPreferences = new GpgPreferences(context);
        }
        return mGpgPreferences;
    }


    private GpgPreferences(Context context) {
        mSharedPreferences = context.getSharedPreferences(PREF_FILE_NAME, 0);
    }


    public void clear() {
        mSharedPreferences.edit().clear().commit();
    }

    public void setGpgOwnPublicKeyId(long keyId) {
        mSharedPreferences.edit().putLong(PREF_PGP_OWN_PUBLIC_KEY_ID, keyId).commit();
    }


    public long getGpgOwnPublicKeyId() {
        return mSharedPreferences.getLong(PREF_PGP_OWN_PUBLIC_KEY_ID, 0);
    }


}

File: crypto/src/main/java/io/oversec/one/crypto/gpg/GpgPreferences.kt
package io.oversec.one.crypto.gpg

import android.annotation.SuppressLint
import android.content.Context

@SuppressLint("CommitPrefEdits")
class GpgPreferences private constructor(context: Context) {
    private val mSharedPreferences = context.getSharedPreferences(PREF_FILE_NAME, 0)

    var gpgOwnPublicKeyId: Long
        get() = mSharedPreferences.getLong(PREF_PGP_OWN_PUBLIC_KEY_ID, 0)
        set(v) {
            mSharedPreferences.edit().putLong(PREF_PGP_OWN_PUBLIC_KEY_ID, v).commit()
        }

    fun clear() {
        mSharedPreferences.edit().clear().commit()
    }

    companion object {

        private const val PREF_PGP_OWN_PUBLIC_KEY_ID = "pgp_own_public_key_id"
        private val PREF_FILE_NAME = GpgPreferences::class.java.simpleName

        @SuppressLint("StaticFieldLeak") // note that we're storing *Application*context
        @Volatile
        private var mGpgPreferences: GpgPreferences? = null

        fun getPreferences(ctx: Context): GpgPreferences =
            mGpgPreferences ?: synchronized(this) {
                mGpgPreferences ?: GpgPreferences(ctx.applicationContext).also { mGpgPreferences = it }
            }

    }
}
